modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -205,6 +205,35 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
       .forall(e => e == RegistryError.NotFound || e.isInstanceOf[RegistryError.ClientFailure])
   }
 
+  /**
+   * Checks if the resolution error indicates a fatal/unrecoverable issue.
+   *
+   * Returns true when:
+   * - Any custom (non-Iglu Central) repository has a RepoFailure or ClientFailure
+   * - All Iglu Central mirrors have a RepoFailure or ClientFailure
+   *
+   * Returns false otherwise.
+   *
+   * If a registry has a fatal error alongside other errors, assume it is unrecoverable.
+   */
+  def isUnrecoverable(error: ResolutionError): Boolean = {
+    val (igluCentral, custom) = error.value.partition { case (repo, _) =>
+      allIgluCentral.contains(repo)
+    }
+
+    def hasFatalError(history: LookupHistory): Boolean =
+      history.errors.exists(e =>
+        e.isInstanceOf[RegistryError.RepoFailure] || e.isInstanceOf[RegistryError.ClientFailure]
+      )
```

Comment:
This is fine.  My personal preference is to do something like:

```scala
history.errors.exists {
  case _: RegistryError.RepoFailure => true
  case _: RegistryError.ClientFailure => true
  case _: RegistryError.NotFound => false
}
```

...because that is more explicit to someone reading the code about how we categorize each of the error types.

But I don't want to nitpick, and other engineers will probably disagree with my preferences!  So how you did it is fine.